benchmark_stream_test.go

```diff
@@ -244,12 +250,48 @@ func TestBenchmarkStreamIO(t *testing.T) {
 			[]byte("1"),
 			[]byte("2"),
 		}
-		var streamIO streamIOBenchmark
+		streamIO := newStreamIOBenchmark(nil, &streamCallResult{})
 
 		for _, response := range responses {
 			require.NoError(t, streamIO.HandleResponse(response))
 		}
 
 		assert.Equal(t, streamIO.streamResponses, responses)
 	})
+
+	t.Run("next request must increment sent msgs counter", func(t *testing.T) {
+		requests := [][]byte{
+			[]byte("1"),
+			[]byte("2"),
+			[]byte("3"),
+		}
+		expectedMsgsSent := 2
+
+		streamResult := &streamCallResult{}
+		streamIO := newStreamIOBenchmark(requests, streamResult)
+
+		for i := 0; i < expectedMsgsSent; i++ {
```

Comment:
Did you do this test on purpose? This test looks so unnatural since you loop on expectedMsgsSent and not on the length of requests. What is the value of doing this?

Reply:
I agree, fixed the test to use messages length